packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.service.js

```diff
@@ -0,0 +1,105 @@
+export default class BillingOrdersPurchasesService {
+  /* @ngInject */
+  constructor($locale, $log, $q, OvhHttp, iceberg) {
+    this.$locale = $locale;
+    this.$log = $log;
+    this.$q = $q;
+    this.OvhHttp = OvhHttp;
+    this.iceberg = iceberg;
+  }
+
+  postPurchaseOrder(data) {
+    return this.OvhHttp.post('/me/billing/purchaseOrder', {
+      rootPath: 'apiv6',
+      data,
+    });
+  }
+
+  putPurchaseOrder(id, data) {
+    return this.OvhHttp.put(`/me/billing/purchaseOrder/${id}`, {
+      rootPath: 'apiv6',
+      data,
+    });
+  }
+
+  getPurchasesOrder() {
+    return this.iceberg('/me/billing/purchaseOrder')
+      .query()
+      .expand('CachedObjectList-Pages')
+      .sort('reference', 'ASC')
+      .limit(5000)
+      .execute(null, true)
+      .$promise.then(({ data }) => data)
+      .then((purchases) =>
+        purchases
+          .filter((elm) => elm.status === 'CREATED')
+          .map((elm) => {
+            const newElm = { ...elm };
+            if (
+              elm.active === true &&
+              this.toDay >= elm.startDate &&
```

Comment:
toDay is a static method, it looks you should do this.constructor.toDay()
also comparing two dates like that won't work, see : https://stackoverflow.com/questions/492994/compare-two-dates-with-javascript
you should use the moment library for this

Reply:
I have move in constructor.
In fact, i compare the string format date (new Date().toISOString().indexOf('T')). 
It works because the compared date is YYYY-MM-DD format.